mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature

```diff
@@ -702,4 +702,56 @@ Feature: Requests tests
     Given path 'circulation', 'requests', extRequestId2
     When method GET
     Then status 200
-    And match $.position == postRequestResponse.response.position
\ No newline at end of file
+    And match $.position == postRequestResponse.response.position
+
+  Scenario: Ensure request queue containing item and title requests is ordered correctly
+    * def extUserId1 = call uuid1
+    * def extUserId2 = call uuid1
+    * def extUserId3 = call uuid1
+    * def extItemId1 = call uuid1
+    * def extItemId2 = call uuid1
+    * def extInstanceId = call uuid1
+    * def extHoldingId = call uuid1
+
+    # enable tlr feature
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@EnableTlrFeature')
+
+    # post users
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostUser') { extUserId: #(extUserId1), extUserBarcode: #('FAT-1508UBC-1'), extGroupId: #(fourthUserGroupId) }
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostUser') { extUserId: #(extUserId2), extUserBarcode: #('FAT-1508UBC-2'), extGroupId: #(fourthUserGroupId) }
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostUser') { extUserId: #(extUserId3), extUserBarcode: #('FAT-1508UBC-3'), extGroupId: #(fourthUserGroupId) }
+
+    # post an instance
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostInstance') { extInstanceId: #(extInstanceId) }
+
+    # post a holding
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostHoldings') { extHoldingsRecordId: #(extHoldingId), extInstanceId: #(extInstanceId) }
+
+    # post the first item and a page tlr
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostItem') { extItemId: #(extItemId1), extItemBarcode: #('FAT-1508IBC-1'), extHoldingsRecordId: #(extHoldingId) }
+    * def extRequestId1 = call uuid1
+    * def tlrResponse = call read('classpath:vega/mod-circulation/features/util/initData.feature@PostTitleLevelRequest') { requestId: #(extRequestId1), requesterId: #(extUserId1), extInstanceId: #(extInstanceId) }
+
+    # post the second item, a page ilr and a hold tlr
+    * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostItem') { extItemId: #(extItemId2), extItemBarcode: #('FAT-1508IBC-2'), extHoldingsRecordId: #(extHoldingId) }
+    * def extRequestId2 = call uuid1
```

Comment:
How about creating first hold tlr and then page ilr, so that no matter the order of creation the queue works as needed?

Reply:
there is not allowed to create hold tlr if there is available item in the instance, so I created page ilr for the second item, then created hold tlr for the instance

Reply:
Exactly, ok